tests-e2e/core/managed_environment_test.go

```diff
@@ -1101,6 +1103,35 @@ var _ = Describe("Environment E2E tests", func() {
 			}
 			Eventually(*managedEnvCR, "2m", "1s").Should(managedenvironment.HaveCredentials(expectedEnvSpec))
 		})
+
+		It("should verify Deletion of GitOpsDeploymentManagedEnvironment on Non-existent Environment", func() {
+			By("creates a GitOpsDeploymentManagedEnvironment with an ownerref to an Environment that doesn't exist")
+			kubeConfigContents, apiServerURL, err := fixture.ExtractKubeConfigValues()
+			Expect(err).To(BeNil())
+
+			managedEnv, _ := buildManagedEnvironment(apiServerURL, kubeConfigContents, true)
+			managedEnv.OwnerReferences = []metav1.OwnerReference{
+				{
+					Kind:       "Environment",
+					Name:       "name",
+					APIVersion: "SomeOtherAPIVersion",
+					UID:        "123",
+				},
+			}
+
+			k8sClient, err = fixture.GetE2ETestUserWorkspaceKubeClient()
+			Expect(err).To(Succeed())
+
+			err = k8s.Create(&managedEnv, k8sClient)
+			Expect(err).To(BeNil())
+
+			Eventually(func() bool {
+				err = k8sClient.Get(ctx, client.ObjectKeyFromObject(&managedEnv), &managedEnv)
+				Expect(apierr.IsNotFound(err)).To(BeTrue())
+				return Expect(err).ToNot(BeNil())
+			}, 1*time.Minute)
```

Comment:
Even easier than doing this, you can just do:
```
			Eventually(managedEnv, "60s", "1s").ShouldNot(k8s.ExistByName(k8sClient))
```

(And in general, I'm not sure if `Expect` statements work (are correct) inside `Eventually`, my guess is they don't, but I haven't actually experimented to find out :thinking: )

Reply:
@jgwest I will update this and `Expect` works as expected in `Eventually`  I have tested in some of the test cases and it worked  , but not sure whether it will work for all types